src/service/resource.js

```diff
@@ -74,8 +74,9 @@ export class Resource {
    * @return {!Resource}
    */
   static forElement(element) {
-    return dev().assert(Resource.forElementOptional(element),
-        'Missing resource prop on %s', element);
+    return /** @type {!Resource} */ (
+        dev().assert(Resource.forElementOptional(element),
```

Comment:
yeah type inference system for asserts dont work with `dev().assert()`. no quick solution yet